yarn-project/end-to-end/src/e2e_rollup_native_asset_contract.test.ts

```diff
@@ -0,0 +1,218 @@
+
+const logger = createDebugLogger('aztec:e2e_rollup_native_asset_contract');
+
+const config = getConfigEnvVars();
+
+const sha256ToField = (buf: Buffer): Fr => {
+  const tempContent = toBigIntBE(sha256(buf));
+  return Fr.fromBuffer(toBufferBE(tempContent % Fr.MODULUS, 32));
+};
+
+describe('e2e_rollup_native_asset_contract', () => {
+  let node: AztecNode;
+  let aztecRpcServer: AztecRPCServer;
+  let account: HDAccount;
+  let accounts: AztecAddress[];
+  let contract: Contract;
+  let portalAddress: EthAddress;
+  let portalContract: any;
+
+  let publicClient: PublicClient<HttpTransport, Chain>;
+  let walletClient: any;
+  let outbox: GetContractReturnType<typeof OutboxAbi, PublicClient<HttpTransport, Chain>>;
+  let registryAddress: Address;
+
+  beforeEach(async () => {
+    account = mnemonicToAccount(MNEMONIC);
+    const privKey = account.getHdKey().privateKey;
+    const {
+      rollupAddress,
+      registryAddress: registryAddress_,
+      outboxAddress,
+      unverifiedDataEmitterAddress,
+    } = await deployL1Contracts(config.rpcUrl, account, logger);
+
+    config.publisherPrivateKey = Buffer.from(privKey!);
+    config.rollupContract = rollupAddress;
+    config.unverifiedDataEmitterContract = unverifiedDataEmitterAddress;
+
+    registryAddress = getAddress(registryAddress_.toString());
+
+    node = await AztecNode.createAndSync(config);
+    aztecRpcServer = await createAztecRpcServer(2, node);
+    accounts = await aztecRpcServer.getAccounts();
+
+    publicClient = createPublicClient({
+      chain: foundry,
+      transport: http(config.rpcUrl),
+    });
+
+    outbox = getContract({
+      address: getAddress(outboxAddress.toString()),
+      abi: OutboxAbi,
+      publicClient,
+    });
+
+    // Deploy L1 portal
+    walletClient = createWalletClient({
+      account,
+      chain: foundry,
+      transport: http(config.rpcUrl),
+    });
+
+    portalAddress = await deployL1Contract(walletClient, publicClient, RollupNativeAssetAbi, RollupNativeAssetBytecode);
+
+    portalContract = getContract({
+      address: getAddress(portalAddress.toString()),
+      abi: RollupNativeAssetAbi,
+      publicClient,
+      walletClient,
+    });
+  }, 60_000);
+
+  afterEach(async () => {
+    await node?.stop();
+    await aztecRpcServer?.stop();
+  });
+
+  const expectBalance = async (owner: AztecAddress, expectedBalance: bigint) => {
+    const ownerPublicKey = await aztecRpcServer.getAccountPublicKey(owner);
+    const [balance] = await contract.methods.getBalance(pointToPublicKey(ownerPublicKey)).view({ from: owner });
+    logger(`Account ${owner} balance: ${balance}`);
+    expect(balance).toBe(expectedBalance);
+  };
+
+  const pointToPublicKey = (point: Point) => {
+    const x = point.buffer.subarray(0, 32);
+    const y = point.buffer.subarray(32, 64);
+    return {
+      x: toBigIntBE(x),
+      y: toBigIntBE(y),
+    };
+  };
+
+  const deployContract = async (initialBalance = 0n, owner = { x: 0n, y: 0n }) => {
```

Comment:
rename to `deployNativeTokenContract`. `deployContract()` is a bit vague (esp since this exact fn name is used across other E2Es too as far as I remember)

Reply:
Its the same thing across the E2Es, part of what is referred to in https://github.com/AztecProtocol/aztec-packages/pull/552#discussion_r1197585771